$NetBSD: patch-ab,v 1.13 2020/07/08 04:06:48 mrg Exp $

Special adaptation for different NetBSD versions and for OpenBSD.
Turn on SYSMALLOC for NetBSD until tc.alloc.c gets better.

--- config_f.h.orig	2019-12-04 09:51:54.000000000 -0800
+++ config_f.h	2020-07-06 15:21:46.122410962 -0700
@@ -47,7 +47,18 @@
  *	         of nls...
  *
  */
-#define SHORT_STRINGS
+#if defined(__NetBSD__)
+# include <sys/param.h>
+# if defined(__NetBSD_Version__) && (__NetBSD_Version__ >= 200000000)
+#  define SHORT_STRINGS
+# else
+#  undef SHORT_STRINGS
+# endif
+#elif defined(__OpenBSD__)
+# undef SHORT_STRINGS
+#else
+# define SHORT_STRINGS
+#endif
 
 /*
  * WIDE_STRINGS	Represent strings using wide characters
@@ -138,7 +149,7 @@
  *		This can be much slower and no memory statistics will be
  *		provided.
  */
-#if defined(__MACHTEN__) || defined(PURIFY) || defined(MALLOC_TRACE) || defined(_OSD_POSIX) || defined(__MVS__) || defined (__CYGWIN__) || defined(__GLIBC__) || defined(__OpenBSD__) || defined(__APPLE__) || defined (__ANDROID__)
+#if defined(__MACHTEN__) || defined(PURIFY) || defined(MALLOC_TRACE) || defined(_OSD_POSIX) || defined(__MVS__) || defined (__CYGWIN__) || defined(__GLIBC__) || defined(__OpenBSD__) || defined(__APPLE__) || defined (__ANDROID__) || defined(__NetBSD__)
 # define SYSMALLOC
 #else
 # undef SYSMALLOC
